src/server/DeviceServer.js

```diff
@@ -327,6 +319,37 @@ class DeviceServer {
     await this._eventPublisher.publish(eventData);
   };
 
+  _onDeviceClaimCodeMessage = async (
+    message: Message,
+    device: Device,
+  ): Promise<void> => {
+    const claimCode = message.getPayload().toString();
+    const deviceID = device.getID();
+    const deviceAttributes = await this._deviceAttributeRepository.getById(deviceID);
+    if (!deviceAttributes) {
+      return;
+    }
+    if (deviceAttributes.ownerID) {
+      return;
+    }
+
+    // todo if we figure out how to delete claimCode from the device
+    // we could make the flow more clean.
+    if (deviceAttributes.claimCode !== claimCode) {
+      const claimRequestUser = await this._userRepository.getByClaimCode(claimCode);
+      if (!claimRequestUser) {
+        return;
+      }
+
+      await this._userRepository.removeClaimCode(claimRequestUser.id, claimCode);
```

Comment:
In my opinion it is not a good option to see which devices the owner has because i can generate multiple claim codes without claiming any device or the claim process can fail and i can restart it creating other claim codes.
If you remove only the ownerId in deviceAttributes you stop the loop of claim on reconnect because deviceAttributes.claimCode === claimCode, or am i wrong?